src/bun.js/bindings/ModuleLoader.cpp

```diff
@@ -350,6 +355,105 @@ extern "C" void Bun__onFulfillAsyncModule(
     promise->resolve(promise->globalObject(), JSC::JSSourceCode::create(vm, JSC::SourceCode(provider)));
 }
 
+JSValue fetchCommonJSModule(
+    Zig::GlobalObject* globalObject,
+    JSValue specifierValue,
+    BunString* specifier,
+    BunString* referrer)
+{
+    void* bunVM = globalObject->bunVM();
+    auto& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    ErrorableResolvedSource resValue;
+    ErrorableResolvedSource* res = &resValue;
+
+    auto& builtinNames = WebCore::clientData(vm)->builtinNames();
+
+    auto synthetic = [specifier, globalObject, referrer](const SyntheticSourceProvider::SyntheticSourceGenerator& generator) -> JSValue {
+        return JSCommonJSModule::create(
+            globalObject,
+            Bun::toWTFString(*specifier),
+            Bun::toWTFString(*referrer),
+            generator);
+    };
+
+    if (Bun__fetchBuiltinModule(bunVM, globalObject, specifier, referrer, res)) {
+        if (!res->success) {
+            throwException(scope, res->result.err, globalObject);
+            RELEASE_AND_RETURN(scope, JSValue());
+        }
+
+        switch (res->result.value.tag) {
+        case SyntheticModuleType::Module: {
+            return synthetic(generateNodeModuleModule);
+        }
+
+        case SyntheticModuleType::Buffer: {
+            return synthetic(generateBufferSourceCode);
+        }
+        case SyntheticModuleType::TTY: {
+            return synthetic(generateTTYSourceCode);
+        }
+        case SyntheticModuleType::NodeUtilTypes: {
+            return synthetic(Bun::generateNodeUtilTypesSourceCode);
+        }
+        case SyntheticModuleType::Process: {
+            return synthetic(generateProcessSourceCode);
+        }
+        case SyntheticModuleType::Events: {
+            return synthetic(generateEventsSourceCode);
+        }
+        case SyntheticModuleType::StringDecoder: {
+            return synthetic(generateStringDecoderSourceCode);
+        }
+        default: {
+            return jsNumber(-1);
+        }
+        }
+    }
+
+    // if (JSC::JSValue virtualModuleResult = JSValue::decode(Bun__runVirtualModule(globalObject, specifier))) {
```

Comment:
regression: loaders don't work for CommonJS here

we can address this though, that feature is sort of already broken